src/main/java/teammates/ui/webapi/DeleteAccountAction.java

```diff
@@ -1,19 +1,34 @@
 package teammates.ui.webapi;
 
+import teammates.common.exception.FirebaseException;
 import teammates.common.util.Const;
+import teammates.common.util.Logger;
 
 /**
  * Action: deletes an existing account (either student or instructor).
+ * <p>The associated Firebase user is also deleted.</p>
  */
 class DeleteAccountAction extends AdminOnlyAction {
 
+    private static final Logger log = Logger.getLogger();
+
     @Override
     public JsonResult execute() {
         String googleId = getNonNullRequestParamValue(Const.ParamsNames.INSTRUCTOR_ID);
         if (fileStorage.doesFileExist(googleId)) {
             fileStorage.delete(googleId);
         }
         logic.deleteAccountCascade(googleId);
+
+        String email = googleId.contains("@") ? googleId : googleId.concat("@gmail.com");
+        try {
+            firebaseInstance.deleteUser(email);
+        } catch (FirebaseException e) {
+            // Deleting Firebase user error logged as warning and not thrown as exception to ensure backwards
+            // compatibility as old TEAMMATES users are not Firebase users
```

Comment:
This is assuming that the exception is because of non-existent user. What if the exception is due to something else?

Reply:
How about having an `errorCode` field in `AuthException` to keep track of the exception cause and do a check in `DeleteAccountAction`. If the exception is due to user not found, log it as a warning(, or do nothing as per the existing implementation of deleting binary files and TEAMMATES account), else throw an exception. The identification of the exception cause is possible as they are retrievable via `FirebaseAuthException.getAuthErrorCode()`.